src/test/java/org/opentripplanner/middleware/GetMonitoredTripsTest.java

```diff
@@ -114,52 +114,59 @@ public static void tearDown() {
         // Expect only 1 trip for solo Otp user.
         assertEquals(1, soloTrips.data.size());
 
         // Get trips for multi Otp user/admin user.
-        HttpResponse<String> multiTripsResponse = mockAuthenticatedGet(MONITORED_TRIP_PATH, multiOtpUser);
-        ResponseList<MonitoredTrip> multiTrips = JsonUtils.getResponseListFromJSON(multiTripsResponse.body(), MonitoredTrip.class);
-
+        ResponseList<MonitoredTrip> multiTrips = getMonitoredTripsForUser(MONITORED_TRIP_PATH, multiOtpUser);
         // Multi Otp user has 'enhanced' admin credentials both trips will be returned. The expectation here is that the UI
         // will always provide the user id to prevent this (as with the next test).
         // TODO: Determine if a separate admin endpoint should be maintained for getting all/combined trips.
         assertEquals(2, multiTrips.data.size());
 
         // Get trips for only the multi Otp user by specifying Otp user id.
-        HttpResponse<String> tripsFilteredResponse = mockAuthenticatedGet(
-            String.format("api/secure/monitoredtrip?userId=%s", multiOtpUser.id),
-            multiOtpUser
+        ResponseList<MonitoredTrip> tripsFiltered = getMonitoredTripsForUser(
+            String.format("%s?userId=%s", MONITORED_TRIP_PATH, multiOtpUser.id), multiOtpUser
         );
-        ResponseList<MonitoredTrip> tripsFiltered = JsonUtils.getResponseListFromJSON(tripsFilteredResponse.body(), MonitoredTrip.class);
         // Just the trip for Otp user 2 will be returned.
         assertEquals(1, tripsFiltered.data.size());
     }
+
+    /**
+     * Helper method to get trips for user.
+     */
+    private ResponseList<MonitoredTrip> getMonitoredTripsForUser(String path, OtpUser otpUser) throws JsonProcessingException {
+        HttpResponse<String> soloTripsResponse = mockAuthenticatedGet(path, otpUser);
+        return JsonUtils.getResponseListFromJSON(soloTripsResponse.body(), MonitoredTrip.class);
+    }
+
+    /**
+     * Creates a {@link MonitoredTrip} for the specified user.
+     */
+    private static void createMonitoredTripAsUser(OtpUser otpUser) throws URISyntaxException {
+        MonitoredTrip monitoredTrip = new MonitoredTrip(TestUtils.sendSamplePlanRequest());
+        monitoredTrip.updateAllDaysOfWeek(true);
+        monitoredTrip.userId = otpUser.id;
+
+        // Set mock OTP responses so that trip existence checks in the
+        // POST call below to save the monitored trip can pass.
+        TestUtils.setupOtpMocks(ItineraryUtilsTest.createMockOtpResponsesForTripExistence());
+
+        HttpResponse<String> createTripResponse = mockAuthenticatedRequest(MONITORED_TRIP_PATH,
+            JsonUtils.toJson(monitoredTrip),
+            otpUser,
+            HttpUtils.REQUEST_METHOD.POST
+        );
+
+        // After POST is complete, reset mock OTP responses for subsequent tests.
+        TestUtils.resetOtpMocks();
```

Comment:
The resetOTPMocks should be moved into an after method so that it always runs even if the test fails.

Reply:
Same remark as https://github.com/ibi-group/otp-middleware/pull/102#discussion_r535584866.